src/handler.rs

```diff
@@ -282,6 +285,18 @@ impl RequestHandler {
 
             let index_files = index_files.as_ref();
 
+            // Check for markdown content negotiation (only if enabled)
+            if self.opts.accept_markdown {
+                if let Some(result) =
+                    crate::markdown::pre_process(&self.opts, req, base_path, req.uri().path())
+                {
+                    let resp = result?;
+                    // Post-process the markdown response (apply CORS, etc.)
+                    let resp = cors::post_process(&self.opts, req, resp)?;
+                    return Ok(resp);
+                }
+            }
```

Comment:
I would suggest instead of handling the file manually in `markdown::pre_process` (trying metadata, reading the whole file, preparing the response, checking CORS, etc), just replace the `uri_path` with the markdown variant and then make sure to pass it over to the `static_files::handle` options.

By doing this, you will let `static_files::handle` do its usual job for you, such as sanitize the path, find the metadata, read the file stream, add proper headers, and prepare the file or error response.
Also, the other post-process functions can be applied later on, like compression, cors, control-headers, security headers, custom headers, etc.

If you need to add extra response headers, you can modify the response later by creating a `post_process` function.

Here is an idea:

```suggestion
            let mut uri_path = req.uri().path();

            // Check for markdown content negotiation (only if enabled)
            if self.opts.accept_markdown {
                if let Some(uri_path_md) =
                    crate::markdown::pre_process(&self.opts, req, base_path, &uri_path)
                {
                    uri_path = uri_path_md;
                }
            }
```